### ISSUE TITLE
ACPI error

### ISSUE THREAD

**Issue by StarTroop** [jhwshin/alx-wol-dkms#1]
I've noticed an issue recently where upon wake using WoL (with this patch) results in an ACPI interrupt that causes high CPU usage through journald.
Journalctl is essentially flooded with these three lines in repeat:  
```
Jul 28 15:36:09 desktop kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_GPE._L09.D1F0], AE_NOT_FOUND (20210331/psargs-330)
Jul 28 15:36:09 desktop kernel: ACPI Error: Aborting method \_GPE._L09 due to previous error (AE_NOT_FOUND) (20210331/psparse-529)
Jul 28 15:36:09 desktop kernel: ACPI Error: AE_NOT_FOUND, while evaluating GPE method [_L09] (20210331/evgpe-511)
```  
I can disable the interrupt manually to stop the messages, but then the alx bug seems to crop up again (immediate wake after first suspend, but then the following suspend seems to stay put.) Using WoL again will simply cause the interrupt to start again.  
I'm not sure if this was the case even when I first installed your fork back when you linked it in the AUR comments, so I want to find out if you were experiencing the same issue.  
A rebuild works fine, the patch completes successfully. I'm using the latest zen kernel (5.13.5), and my mobo is an MSI z87-gd65 (I'm wondering if this is specific to my hardware.)

**Comment by StarTroop** [comment]
Kernel parameter `acpi_mask_gpe=0x6f` solves the cpu abuse issue, but the buggy behaviour still persists (first suspend after a wakeup with WoL will immediately wake up again, but following suspends are normal). I can live with it in this state, though.

**Comment by jshin137** [comment]
I am getting the same error message as you mentioned in `journalctl` when using WoL:
```
Jul 30 15:18:05 desktop kernel: ACPI BIOS Error (bug): Could not resolve symbol [\_GPE._L09.D1F0], AE_NOT_FOUND (20210331/psargs-330)
Jul 30 15:18:05 desktop kernel: ACPI Error: Aborting method \_GPE._L09 due to previous error (AE_NOT_FOUND) (20210331/psparse-529)
Jul 30 15:18:05 desktop kernel: ACPI Error: AE_NOT_FOUND, while evaluating GPE method [_L09] (20210331/evgpe-511)
```
However, I am not experiencing high CPU usage nor is it flooding my journalctl.
I have tried WoL from power off state and from suspend state (after using `systemctl suspend` to replicate your issue multiple times). I am also not experiencing the WoL bug that causes immediate wakes from the mentioned states.

I am currently running the latest Arch kernel `5.13.6-arch1-1` on mobo `Gigabyte Z97X-SOC Force`.

**Comment by StarTroop** [comment]
I've experimented with the LTS kernel and removing some other ACPI-related parameters from my loader (to fix a separate display-waking issue), but still get the issue. Most likely it's just my specific hardware, though even if it'd work using the vanilla kernel, I don't want to resort to that so I'm happy to leave it as-is. It's worth pointing out that your mobo uses the E2200 LAN controller while mine is the E2205. Don't know the technical differences but it probably accounts for the discrepancy in our experiences.

**Comment by jshin137** [comment]
Not sure if it is relevant to your issue, but I remember now that I did have an issue with waking up randomly even after the updated patch.

`sudo ethtool <NET_INTERFACE>`

```
Settings for <INTERFACE>:
	Supported ports: [ TP ]
        ...
	Supports Wake-on: pg
	Wake-on: g
    Current message level: 0x000060e4 (24804)
                               link ifup rx_err tx_err hw wol
	Link detected: yes
```

As shown settings above originally it was:

`Wake-on: p`

which I changed to:

`Wake-on: g`

using:

`sudo ethtool -s <NET_INTERFACE> wol g`

as previous setting caused any activity in network device to wake even without a magic packet. (i.e any packet sent to network device)

`man ethtool`
```
...
           wol p|u|m|b|a|g|s|f|d...
                  Sets Wake-on-LAN options.  Not all devices support this.  The argument to this  option  is  a  string  of
                  characters specifying which options to enable.

                  p   Wake on PHY activity
                  u   Wake on unicast messages
                  m   Wake on multicast messages
                  b   Wake on broadcast messages
                  a   Wake on ARP
                  g   Wake on MagicPacket™
                  s   Enable SecureOn™ password for MagicPacket™
                  f   Wake on filter(s)
                  d   Disable  (wake  on  nothing).  This option
                      clears all previous options.
...
```

Otherwise, my next guess might be a USB device causing it to wake?

`cat /proc/acpi/wakeup`

**Comment by StarTroop** [comment]
I initially used Network Manager to set the correct mode, but I already did disable the PHY activity. Just now I used ethtool to double check, which confirms that while both "p" and "g" are supported, only "g" was enabled. I didn't suspect this was the issue since in Windows I had initially misconfigured WoL by allowing the LAN device to wake the computer, but forgetting to limit it to magic packets. However, the behaviour in that instance was different as it would only wake up occasionally (probably like when you had PHY enabled). In my current instance, the computer only wakes up once immediately after the first suspend after using WoL, then it's back to normal.
